imports/plugins/core/orders/client/containers/orderDashboardContainer.js

```diff
@@ -194,6 +260,397 @@ class OrderDashboardContainer extends Component {
+    if (ordersToRegress) {
+      this.displayRegressionAlert(
+        selectedOrders, ordersToRegress, status, whichFalseState, isNotPacked, isNotLabeled, isLabeled
+      );
+
+      // display proper alert if the order(s) are in this state already or want to skip the previous states
+    } else this.displayAlert(selectedOrders, status, whichFalseState, isNotPacked, isNotLabeled, isLabeled);
+  }
+
+
+  shippedShippingStatus = (selectedOrders, status) => {
+    let isNotLabeled = 0;
+    let isNotShipped = 0;
+    let isShipped = 0;
+    let whichFalseState = "";
+
+    selectedOrders.forEach((order) => {
+      // check if the order(s) are in this state already or in one of the previous states
+      if (order.shipping[0].workflow.status === "new") {
+        isNotLabeled++;
+        whichFalseState = shippingStates.picked;
+      } else if (order.shipping[0].workflow.status === "coreOrderWorkflow/picked") {
+        isNotLabeled++;
+        whichFalseState = shippingStates.packed;
+      } else if (order.shipping[0].workflow.status === "coreOrderWorkflow/packed") {
+        isNotLabeled++;
+        whichFalseState = shippingStates.labeled;
+      } else if (order.shipping[0].workflow.status === "coreOrderWorkflow/labeled") {
+        isNotShipped++;
+      } else if (order.shipping[0].workflow.status === "coreOrderWorkflow/shipped") {
+        isShipped++;
+      }
+    });
+
+    // display proper alert if the order(s) are in this state already or want to skip the previous states
+    this.displayAlert(selectedOrders, status, whichFalseState, isNotLabeled, isNotShipped, isShipped);
+  }
+
+  /**
+   * setShippingStatus
+   *
+   * @summary call the relevant method based on the provided shipping status
+   * @param {String} status - the selected shipping status to be set
+   * @param {Array} selectedOrdersIds - array of ids of the selected orders
+   * @return {null} no return value
+   */
+  setShippingStatus = (status, selectedOrdersIds) => {
+    this.setState({
+      renderFlowList: true
+    });
+    const selectedOrders = this.state.orders.filter((order) => {
+      return selectedOrdersIds.includes(order._id);
+    });
+
+    if (status === "picked") {
+      this.pickedShippingStatus(selectedOrders, status);
+    }
+
+    if (status === "packed") {
+      this.packedShippingStatus(selectedOrders, status);
+    }
+
+    if (status === "labeled") {
+      this.labeledShippingStatus(selectedOrders, status);
+    }
+
+    if (status === "shipped") {
+      this.shippedShippingStatus(selectedOrders, status);
+    }
+  }
+
+  handleBulkPaymentCapture = (selectedOrdersIds) => {
+    this.setState({
+      isLoading: {
+        capturePayment: true
+      }
+    });
+    const selectedOrders = this.state.orders.filter((order) => {
+      return selectedOrdersIds.includes(order._id);
+    });
+
+    let orderCount = 0;
+
+    selectedOrders.forEach((order) => {
+      Meteor.call("orders/approvePayment", order, (err) => {
```

Comment:
We should be batching these and performing a single method call rather than making a call for each order